src/main/webapp/resources/js/pages/projects/samples-metadata-import/redux/importReducer.ts

```diff
@@ -91,79 +95,173 @@ export const saveMetadata = createAsyncThunk<
+          )
+          .map(([key, value]) => ({
+            field: key,
+            value,
+            restriction: headers.filter((header) => header.name === key)[0]
+              .targetRestriction,
+          }));
+        return { name, sampleId, metadata: metadataFields };
+      });
+    //create a request with a list of project samples to be updated
+    if (updateSampleList.length > 0) {
+      const updateSampleListChunkSize = calculateChunkSize(
+        updateSampleList.length
+      );
+      const chunkedUpdateSampleList = chunkArray(
+        updateSampleList,
+        updateSampleListChunkSize
+      );
+      for (const chunk of chunkedUpdateSampleList) {
+        updateSamplesPromises.push(
+          updateSamples({
+            projectId,
+            body: chunk,
+          })
+            .then((response) => {
+              const { responses } = response.data;
+              Object.keys(responses).map((key) => {
+                const { error, errorMessage } = responses[key];
+                metadataSaveDetails[key] = {
+                  saved: !error,
+                  error: errorMessage,
+                };
+              });
+            })
+            .catch((error) => {
+              const { responses } = error.response.data;
+              Object.keys(responses).map((key) => {
+                const { error, errorMessage } = responses[key];
+                metadataSaveDetails[key] = {
+                  saved: !error,
+                  error: errorMessage,
+                };
+              });
+            })
         );
+      }
+      //send multiple update project sample requests in parallel
+      const updateSamplesPromiseChunkSize = calculateChunkSize(
+        updateSamplesPromises.length
+      );
+      const chunkedUpdateSamplesPromises = chunkArray(
+        updateSamplesPromises,
+        updateSamplesPromiseChunkSize
+      );
+      for (const chunk of chunkedUpdateSamplesPromises) {
+        await Promise.all(chunk).then(() => {
+          dispatch(
+            setMetadataSaveDetails(Object.assign({}, metadataSaveDetails))
+          );
+        });
+      }
+    }
+
+    //create new project samples
+    const createSamplesPromises: Promise<void>[] = [];
+    const createSampleList = selectedSampleList
+      .filter((metadataItem) => {
+        const name = metadataItem[sampleNameColumn];
+        const sampleId = metadataValidateDetails[name].foundSampleId;
+        return !sampleId;
+      })
+      .map((metadataItem) => {
+        const name = metadataItem[sampleNameColumn];
+        const metadataFields = Object.entries(metadataItem)
+          .filter(
+            ([key]) =>
+              headers.map((header) => header.name).includes(key) &&
+              key !== sampleNameColumn
+          )
+          .map(([key, value]) => ({
+            field: key,
+            value,
+            restriction: headers.filter((header) => header.name === key)[0]
```

Comment:
Same as comment above. Why are we sending the restriction here when that is already done on line 90.

Reply:
I forgot to remove old code. Please see eb581b8501124b4ffcc6d213e8bd6624ac57e7be.